common/src/main/java/org/candlepin/common/resteasy/interceptor/DynamicFilterInterceptor.java

```diff
@@ -48,30 +48,38 @@
     @Override
     public ServerResponse preProcess(HttpRequest request, ResourceMethod method)
         throws Failure, WebApplicationException {
+
         Map<String, List<String>> queryParams = request.getUri().getQueryParameters();
-        boolean containsExcl = queryParams.containsKey("exclude");
-        boolean containsIncl = queryParams.containsKey("include");
-        // Cannot do both types of filtering together,
-        // no point in continuing if neither are present
-        if ((containsExcl && containsIncl) ||
-            !(containsExcl || containsIncl)) {
-            return null;
+        boolean containsExcludes = queryParams.containsKey("exclude");
+        boolean containsIncludes = queryParams.containsKey("include");
+
+        DynamicFilterData filterData = new DynamicFilterData();
+
+        if (queryParams.containsKey("filtermode")) {
+            List<String> values = queryParams.get("filtermode");
+            filterData.setWhitelistMode("whitelist".equalsIgnoreCase(values.get(0)));
+        } else {
```

Comment:
[checkstyle] /home/wpoteat/candlepin/common/src/main/java/org/candlepin/common/resteasy/interceptor/DynamicFilterInterceptor.java:61:9: '}' should be alone on a line.